dist/tools/pioasm/patches/riot_pioasm.patch

```diff
@@ -0,0 +1,191 @@
++#include <iostream>
++#include <cinttypes>
++#include "output_format.h"
++#include "pio_disassembler.h"
++
++struct riot_output : public output_format {
++    struct factory {
++        factory() {
++            output_format::add(new riot_output());
++        }
++    };
++
++    riot_output() : output_format("RIOT") {}
++
++    std::string get_description() override
++    {
++        return "C header suitable for use with RIOT-OS";
++    }
++
++    void output_symbols(FILE *out, std::string prefix, const std::vector<compiled_source::symbol> &symbols)
++    {
++        int count = 0;
++        for (const auto &s : symbols) {
++            if (!s.is_label) {
++                fprintf(out, "#define %s%s %d\n", prefix.c_str(), s.name.c_str(), s.value);
++                count++;
++            }
++        }
++        fprintf(out, count ? "\n" : "");
++        count = 0;
++        for (const auto &s : symbols) {
++            if (s.is_label) {
++                fprintf(out, "#define %sLABEL_%s %d\n", prefix.c_str(), s.name.c_str(), s.value);
++                count++;
++            }
++        }
++        fprintf(out, count ? "\n" : "");
++    }
++
++    int output(std::string destination, std::vector<std::string> output_options,
++               const compiled_source &source) override
++    {
++        for (const auto &program : source.programs) {
++            for(const auto &p : program.lang_opts) {
++                if (p.first.size() >= name.size() && p.first.compare(0, name.size(), name) == 0) {
++                    std::cerr << "warning: " << name << " does not support output options; " << p.first << " lang_opt ignored.\n";
++                }
++            }
++        }
++        FILE *out = open_single_output(destination);
++        if (!out) {
++            return 1;
++        }
++        std::string fname = destination.substr(destination.find_last_of("/\\") + 1);
++                    fname = fname.substr(0, std::max(fname.find_first_of("."), (std::size_t)0));
++        std::transform(fname.begin(), fname.end(), fname.begin(), ::toupper);
++        fprintf(out, "#ifndef PIO_%s_H\n", fname.c_str());
++        fprintf(out, "#define PIO_%s_H\n", fname.c_str());
++        fprintf(out, "\n"
++                     "#include <stdbool.h>\n"
++                     "#include \"periph/pio.h\"\n");
++        fprintf(out, "\n"
++                     "#ifdef __cplusplus\n"
++                     "extern \"C\" {\n"
++                     "#endif\n");
++        fprintf(out, "\n");
++        output_symbols(out, "", source.global_symbols);
++
++        for (const auto &program : source.programs) {
++            std::string prefix = "PIO_" + program.name + "_";
++            std::transform(prefix.begin(), prefix.end(), prefix.begin(), ::toupper);
++            fprintf(out, "/* %s */\n", program.name.c_str());
++            fprintf(out, "#define %sSIDESET_COUNT       %d\n", prefix.c_str(), program.sideset_bits_including_opt.value - !!program.sideset_opt);
```

Comment:
if pio does not use a .side_set 0 opt in case of no side used this is:
#define PIO_FRAME_SIDESET_COUNT       32766